src/System.Management.Automation/engine/Modules/ModuleCmdletBase.cs

```diff
@@ -323,6 +323,46 @@ internal List<WildcardPattern> MatchAll
                 }
             }
 
+            if (!found)
```

Comment:
If it is only for Unix maybe use "#if UNIX"?

Reply:
it's technically any filesystem that's case sensitive and technically (although unlikely), it seems you can make NTFS case sensitive.  for practical cases, it's probably fine to #ifdef this and if a customer has an issue on Windows, we can revisit

Reply:
fixed and added comment